guides/plugins/plugins/checkout/cart/add-cart-discounts.md

```diff
@@ -0,0 +1,128 @@
+
+## Creating the processor
+
+To add a discount to the cart, you should use the processor pattern. For this you need to create your own cart processor.
+We'll start with creating a new class called `ExampleProcessor` in the directory `<plugin root>/src/Core/Checkout`. 
+Our class has to implement `Shopware\Core\Checkout\Cart\CartProcessorInterface` and we have to inject `Shopware\Core\Checkout\Cart\Price\PercentagePriceCalculator` in our constructor.
+All adjustments are done in the `process` method, where the product items already own a name and a price.
+
+Let's start with the actual example code:
+
+{% code title="<plugin root>/src/Core/Checkout/ExampleProcessor.php" %}
+```php
+<?php declare(strict_types=1);
+
+namespace Swag\BasicExample\Core\Checkout;
+
+use Shopware\Core\Checkout\Cart\Cart;
+use Shopware\Core\Checkout\Cart\CartBehavior;
+use Shopware\Core\Checkout\Cart\CartProcessorInterface;
+use Shopware\Core\Checkout\Cart\LineItem\CartDataCollection;
+use Shopware\Core\Checkout\Cart\LineItem\LineItem;
+use Shopware\Core\Checkout\Cart\LineItem\LineItemCollection;
+use Shopware\Core\Checkout\Cart\Price\PercentagePriceCalculator;
+use Shopware\Core\Checkout\Cart\Price\Struct\PercentagePriceDefinition;
+use Shopware\Core\Checkout\Cart\Rule\LineItemRule;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+
+class ExampleProcessor implements CartProcessorInterface
+{
+    /**
+     * @var PercentagePriceCalculator
+     */
+    private $calculator;
+
+    public function __construct(PercentagePriceCalculator $calculator)
+    {
+        $this->calculator = $calculator;
+    }
+
+    public function process(CartDataCollection $data, Cart $original, Cart $toCalculate, SalesChannelContext $context, CartBehavior $behavior): void
+    {
+        $products = $this->findExampleProducts($toCalculate);
+
+        // no example products found? early return
+        if ($products->count() === 0) {
+            return;
+        }
+
+        $discountLineItem = $this->createDiscount('EXAMPLE_DISCOUNT');
+
+        // declare price definition to define how this price is calculated
+        $definition = new PercentagePriceDefinition(
+            -10,
+            new LineItemRule(LineItemRule::OPERATOR_EQ, $products->getKeys())
+        );
+
+        $discountLineItem->setPriceDefinition($definition);
+
+        // calculate price
+        $discountLineItem->setPrice(
+            $this->calculator->calculate($definition->getPercentage(), $products->getPrices(), $context)
+        );
+
+        // add discount to new cart
+        $toCalculate->add($discountLineItem);
+    }
+
+    private function findExampleProducts(Cart $cart): LineItemCollection
+    {
+        return $cart->getLineItems()->filter(function (LineItem $item) {
+            if ($item->getType() !== LineItem::PRODUCT_LINE_ITEM_TYPE) {
```

Comment:
Maybe add a comment here, such as `// Only consider products, not custom line items or promotional line items